tests/integration/test_data/gomod_1.18_e2e_test/bom.json

```diff
@@ -1436,6 +1436,17 @@
       "purl": "pkg:golang/regexp?type=package",
       "type": "library"
     },
+    {
```

Comment:
First of all, this should have been part of  the previous commit if it introduced this breakage. Secondly,

> Probably, by adding cargo to cachi2 image some test data (SBOMs) must
> be updated, otherwise CI won't pass.

^this isn't a very descriptive message, there's too much uncertainty in it and not many facts. The real reason behind this needed update are cargo's direct dependencies, more specifically the `glibc-devel` package which provides the object (i.e. .so) standard C libraries.

Lastly, if we manage to install cargo in a clever manner we shouldn't need this test data update just yet.